lib/net/ldap/connection.rb

```diff
@@ -36,7 +36,7 @@ def prepare_socket(server)
     encryption = server[:encryption]
 
     @conn = socket
-    setup_encryption encryption if encryption
+    setup_encryption({ verify_host: server[:connected_host] }.merge(encryption)) if encryption
```

Comment:
@tarcieri, thanks, got it now. I cloned your fork and tested `:verify_host => false` (with valid variables set for my env). It bumps into an error.

Can you provide some config examples?

```ruby
  conn = Net::LDAP.new :host => hostname,
                       :port => 636,
                       :base => base,
                       :encryption => { :method => :simple_tls,
                                        :tls_options => { :verify_mode => OpenSSL::SSL::VERIFY_PEER,
                                                          :verify_host => false,
                                                          :ca_file => ca_file } },
                       :auth => { :username => login,
                                  :password => pass,
                                  :method => :simple }
  conn.bind 
/usr/lib/ruby/2.1.0/openssl/ssl.rb:87:in `block in set_params': undefined method `verify_host=' for #<OpenSSL::SSL::SSLContext:0x000000016f4130> (NoMethodError)
	from /usr/lib/ruby/2.1.0/openssl/ssl.rb:87:in `each'
	from /usr/lib/ruby/2.1.0/openssl/ssl.rb:87:in `set_params'
	from /home/a211278l/Own Tech Examples/Ruby/LDAPS/ruby-net-ldap-fix2/lib/net/ldap/connection.rb:86:in `wrap_with_ssl'

```